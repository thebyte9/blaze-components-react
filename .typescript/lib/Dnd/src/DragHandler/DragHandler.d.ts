import React from "react";
interface IDragHandlerProps {
    onDragStart: (...args: any[]) => any;
}
declare const DragHandler: React.SFC<IDragHandlerProps>;
export default DragHandler;
