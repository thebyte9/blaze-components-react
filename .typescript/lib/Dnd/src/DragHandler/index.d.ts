import DragHandler from "./DragHandler";
export default DragHandler;
