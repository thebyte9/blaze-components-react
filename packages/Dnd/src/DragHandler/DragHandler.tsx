import React from 'react';
interface IDragHandlerProps {
  onDragStart: (...args: any[]) => any;
  isDraggable: boolean
}
const DragHandler: React.SFC<IDragHandlerProps> = ({ onDragStart, isDraggable }) => (
  <div className="list__drag" onDragStart={onDragStart} draggable={isDraggable} data-testid="drag-handler">
    <i className="material-icons">{isDraggable ? 'drag_indicator' : 'drag_disabled'}</i>
  </div>
);
export default DragHandler;