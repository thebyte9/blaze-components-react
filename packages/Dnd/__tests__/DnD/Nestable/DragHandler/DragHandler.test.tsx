import { cleanup, render } from "@testing-library/react";
import expect from "expect";
import React from "react";
import DragHandler from "../../../../src/DragHandler";

describe("Drag handler", () => {
  afterEach(cleanup);
  it("should be defined", () => {
    expect(DragHandler).toBeDefined();
  });

  it("should render withouth throwing error", () => {
    const { container } = render(<DragHandler onDragStart={jest.fn} />);
    expect(container).toMatchSnapshot();
  });
});
