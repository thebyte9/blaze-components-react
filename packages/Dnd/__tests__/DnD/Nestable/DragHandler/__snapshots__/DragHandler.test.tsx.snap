// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drag handler should render withouth throwing error 1`] = `
<div>
  <div
    class="list__drag"
    data-testid="drag-handler"
    draggable="true"
  >
    <i
      class="material-icons"
    >
      drag_indicator
    </i>
  </div>
</div>
`;
